### PR TITLE
Add TRTServer options for setting priority and request timeout

### PR DESCRIPTION
**Pull request by GuanLuo**
This PR is branch-off from `gluo-timeout` and targeted to `gluo-timeout` to show the change for adding TRTServer options on top of the changes in `gluo-timeout`

### CODE REVIEW

File: src/core/trtserver.cc [hunk 1]
```diff
@@ -484,6 +486,22 @@ TrtServerRequestOptions::SetBatchSize(uint64_t batch_size)
...
+  std::lock_guard<std::mutex> lk(mtx_);
+  request_header_->set_priority(priority);
+  return nullptr;  // Success
+}
+
+TRTSERVER_Error*
+TrtServerRequestOptions::SetTimeoutMs(uint64_t timeout_ms)
+{
+  std::lock_guard<std::mutex> lk(mtx_);
+  request_header_->set_timeout_microseconds(timeout_ms);
```
Comment (nieksand) [inline src/core/trtserver.cc]: With `ms` I'd assume milliseconds.  Consider using `us` if it's really microseconds.

File: src/core/ensemble_scheduler.cc [hunk 2]
```diff
@@ -555,6 +557,9 @@ EnsembleContext::InitStep(size_t step_idx, std::shared_ptr<Step>* step)
   irequest->SetCorrelationId(correlation_id_);
   irequest->SetFlags(flags_);
   irequest->SetBatchSize((batch_size == 0 ? 1 : batch_size));
+  irequest->SetPriority(priority_);
+  // FIXME: need to think about how to interpret timeout in ensemble.
```
Comment (deadeyegoodwin) [inline src/core/ensemble_scheduler.cc]: I would say "Request for ensemble model cannot override the timeout values for the composing models."
To support an overall ensemble timeout we would have to add a timeout to the ModelEnsembling section of the model config and then check that timeout between each pass of the ensemble. Can you file a ticket for that.

File: src/core/trtserver.cc [hunk 3]
```diff
@@ -484,6 +486,22 @@ TrtServerRequestOptions::SetBatchSize(uint64_t batch_size)
...
+TRTSERVER_Error*
+TrtServerRequestOptions::SetPriority(uint32_t priority)
+{
+  std::lock_guard<std::mutex> lk(mtx_);
+  request_header_->set_priority(priority);
+  return nullptr;  // Success
+}
+
+TRTSERVER_Error*
+TrtServerRequestOptions::SetTimeoutMs(uint64_t timeout_ms)
```
Comment (deadeyegoodwin) [inline src/core/trtserver.cc]: SetTimeoutMicroseconds

File: src/core/trtserver.cc [hunk 4]
```diff
@@ -980,6 +998,26 @@ TRTSERVER_InferenceRequestOptionsSetBatchSize(
...
+    TRTSERVER_InferenceRequestOptions* request_options, uint32_t priority)
+{
+  TrtServerRequestOptions* loptions =
+      reinterpret_cast<TrtServerRequestOptions*>(request_options);
+  loptions->SetPriority(priority);
+  return nullptr;  // Success
+}
+
+TRTSERVER_Error*
+TRTSERVER_InferenceRequestOptionsSetTimeout(
```
Comment (deadeyegoodwin) [inline src/core/trtserver.cc]: SetTimeoutMicroseconds

File: src/core/ensemble_scheduler.cc (hunk 2 above)
Comment (GuanLuo) [reply src/core/ensemble_scheduler.cc]: The comment is added to #1109 where the request timeout is defined (`api.proto` )

**Pull request merged**